src/contributing.md

```diff
@@ -0,0 +1,140 @@
+# Contributing guidelines
+
+## Generic Reusable Component Design
+
+Define the component's purpose. Narrow down the scope as much as possible.
+
+Avoid assumptions about its parent/owner.
+
+Define the interface/contract. Make explicit what characteristics can be set by the owner and which are read-only or undefined behavior. (prefix with `_` private internal properties?)
+
+If control maps to a QML control, use QML2's customization mechanisms. [Check docs for examples](https://doc.qt.io/qt-6/qtquickcontrols2-customize.html)
```

Comment:
I meant `B`: creating a custom control should be first considered using the QML2's way as it can be seen in the documentation from the link provided. That means replacing defaults with custom items like markers, popups, backgrounds, etc.

This recommendation advocates against starting to build one from scratch without considering the alternatives. Most of the time the cost of creating complex controls and the effort to cover all the corner cases is high. QML2 provides the basics, and make sense to use it as long as the design follows the basic rules enforced by the QML's mechanism.

> we should use Qt controls instead of defining our own unless we have a very good reason to do that.
I integrated this great suggestion, thanks.